src/planner/rules/plan.rs

```diff
@@ -582,6 +583,47 @@ mod tests {
         rules
     }
 
+    egg::test_fn! {
+        cancel_limit,
+        rules(),
+        // SELECT name
+        // FROM student
+        // WHERE true
+        // LIMIT 0
+        "
+        (proj (list $1.2)
+        (limit null 0
+            (filter true
+                (scan $1 (list $1.1 $1.2) null)
+            )
+        ))" => "
+        (proj
+            (list $1.2)
+            (scan $1 (list $1.1 $1.2) null))
+        "
+    }
+
+    egg::test_fn! {
+        merge_filter,
+        rules(),
+        // SELECT name
+        // FROM student
+        // LIMIT 10
+        // Order by name
+        "
+        proj (list $1.2)
+        (limit 10 0
+            (order (list $1.2) 
+                (scan $1 (list $1.1 $1.2) null)
+            )
+        )" => "
+        proj (list $1.2)
+        （topn 10 0 (list) 
```

Comment:
But why this test can pass?

```suggestion
        （topn 10 0 (list $1.2) 
```

Reply:
I think this is a bug and I will fix it

Reply:
cli failed.
Why?

Reply:
Permission denied. It's a random happened failure. Let's retry it.